compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -326,6 +335,8 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
                   super.transform(_).asInstanceOf[Template]))
           }
         case tree: ValDef =>
+          if tree.symbol.is(Inline, butNot = Param) then
+            ctx.compilationUnit.needsInlining = true
```

Comment:
It's not clear to me why we need this.

Reply:
Now `Inlining` checks that the rhs of inline vals are constants. Technically we could do this in a later phase.